packages/compass-connections/src/components/non-genuine-connection-modal.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {
+  css,
+  Banner,
+  Link,
+  spacing,
+  Body,
+  BannerVariant,
+  showConfirmation,
+} from '@mongodb-js/compass-components';
+import {
+  getConnectionTitle,
+  type ConnectionInfo,
+} from '@mongodb-js/connection-info';
+
+const modalBodyStyles = css({
+  marginTop: spacing[3],
+  marginBottom: spacing[2],
+});
+
+export function showNonGenuineMongoDBWarningModal(
+  connectionInfo?: ConnectionInfo
+) {
+  return showConfirmation({
+    title: 'Non-Genuine MongoDB Detected',
+    description: (
+      <>
+        <Banner variant={BannerVariant.Warning}>
+          {connectionInfo
+            ? `Server or service "${getConnectionTitle(connectionInfo)}"`
```

Comment:
This modal wasn't taking multiple connections into account anymore, I changed the text a bit to be explicit about for which connection we show the modal